src/components/Header/Header.tsx

```diff
@@ -0,0 +1,76 @@
+'use client'
```

Comment:
Does this ensure the whole component is rendered on the Client side?

Reply:
useState requires the component to be the client. 
For more information: [https://nextjs.org/docs/app/building-your-application/rendering/client-components](url)